### DISCUSSION TITLE
Quasar Blocks?

### DISCUSSION THREAD

**Discussion by tohagan** [quasarframework/quasar#15376] [Ideas / Proposals]
PrimeVue offers an interesting suite of pre-assembled components that it calls [PrimeBlocks](https://blocks.primevue.org/#/). 

I wonder if you'd consider add this whole new dimension to Quasar's roadmap?  

One example block on my wish is something like VSCode's Preferences view that allow us to rapidly create a user or system level Settings component.  Lots of the sample code in Quasar examples are already Blocks of one kind or another (e.g. Card examples).  PrimeBlocks list would give you plenty of ideas for more.  

Many of the PrimeBlocks are targeted at creating business or ecommerce web sites whereas I think Quasar's multi platform strength is more about building applications, but both are fertile grounds for innovation in this space.  Watching the PrimeBlock demo made me think they used it a bit more like copy/pasting chunks of (often repeating) code reminiscent of the Boostrap templates you can buy. This is not really what good Vue coding is about, that should be more data driven with recurring components built from data arrays and objects.   So I'd envision Quasar Blocks would accept high level prop objects (TypeScript interfaces?) or optionally a store that configure the internal components. You might disagree.

**Comment by dietrichg** [comment]
I'd be interested in developing this if no one else has a solution. I find myself creating very similar code or layouts sometimes and it would be nice to have a library to choose from.
